pxtlib/service.ts

```diff
@@ -680,8 +686,11 @@ namespace ts.pxtc {
             })))
             .then(() => cleanLocalizations(apis))
             .finally(() => {
-                if (Object.keys(errors).length)
-                    pxt.reportError(`loc.errors`, `invalid translation`, errors);
+                if (errors.length)
+                    pxt.reportError(`loc.errors`, `invalid translations`, {
+                        blockIds: errors.join("|"),
```

Comment:
Instead of this, send multiple reportError with each reportErro to have a single blockId. You can do this in a loop. That will be way easier to index in _by_ query.